### ISSUE TITLE
Combining diacritical marks are shifted with letterSpacing > 0.0 in rendered text

### ISSUE THREAD

**Issue by tgucio** [flutter/flutter#118127]
## Steps to Reproduce

1. Execute `flutter run` on the code sample

**Expected results:** Diacritics are stacked above their letters

**Actual results:** Diacritics are shifted for `TextStyle` with `letterSpacing` > 0.0.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    const TextStyle style = TextStyle(fontSize: 16.0);
    return Scaffold(
      appBar: AppBar(title: Text(widget.title)),
      body: Center(
        child: Column(
          children: <Widget>[
            const SizedBox(height: 100.0),
            Text('Zalgo', style: style.copyWith(letterSpacing: 0.0)),
            const SizedBox(height: 100.0),
            Text('Zalgo', style: style.copyWith(letterSpacing: 1.0)),
            const SizedBox(height: 100.0),
            Text('Zalgo', style: style.copyWith(letterSpacing: 2.0)),
            const SizedBox(height: 100.0),
          ],
        ),
      ),
    );
  }
}
```

</details>

**Comment by danagbemava-nc** [comment]
<img width="400" alt="Screenshot 2023-01-09 at 06 30 43" src="https://user-images.githubusercontent.com/88313112/211251787-a54ab866-1c13-4d6c-b3e9-e3be2e751046.png">

Hi @tgucio, above is what I see what I run the code sample provided above. Is there something else I need to do to reproduce the issue or does the code sample need to be updated? 
Also, what platform(s) are you experiencing this issue on?

Thank you

**Comment by tgucio** [comment]
@danagbemava-nc Looks like the combining diacritics have been removed by github on paste. Let me try again.

**Comment by tgucio** [comment]
@danagbemava-nc github seems to be filtering out these characters now. I updated the description to paste "Zalgo" with combining diacritics from elsewhere.

**Comment by danagbemava-nc** [comment]
Issue is reproducible using the code sample provided. 

I tested on `android`, `iOS`, `mac`, `canvaskit` & `HTML` and only `HTML` seemed to render it properly. The rest of the platforms reproduced the issue. 

<details>
<summary>screenshots</summary>

### macOS
<img width="330" alt="Screenshot 2023-01-10 at 06 07 33" src="https://user-images.githubusercontent.com/88313112/211478545-f891c77e-8984-403e-b64d-efddef5f99c2.png">

### iOS
<img src="https://user-images.githubusercontent.com/88313112/211478559-9b9fd983-87ca-41bc-abda-63393a318514.png" width="300" />

### HTML
<img width="300" alt="Screenshot 2023-01-10 at 06 16 50" src="https://user-images.githubusercontent.com/88313112/211478568-c75bff58-a7bd-455b-bc50-92366b405b29.png">


### Canvaskit
<img width="300" alt="Screenshot 2023-01-10 at 06 29 16" src="https://user-images.githubusercontent.com/88313112/211478574-c6e3fca1-1282-41c2-a2d6-880dc9b796bd.png">

### Android
<img src="https://user-images.githubusercontent.com/88313112/211478595-1df07788-22b0-4083-a109-9de931581db7.jpeg" />

</details>

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    const TextStyle style = TextStyle(fontSize: 16.0);
    return Scaffold(
      appBar: AppBar(title: Text(widget.title)),
      body: Center(
        child: Column(
          children: <Widget>[
            const SizedBox(height: 100.0),
            Text('Z͉̳̺ͥͬ̾a̴͕̲̒̒͌̋ͪl̨͎̰̘͉̟ͤ̀̈̚͜g͕͔̤͖̟̒͝ͅo̵̡̡̼͚̐ͯ̅ͪ̆ͣ̚', style: style.copyWith(letterSpacing: 0.0)),
            const SizedBox(height: 100.0),
            Text('Z͉̳̺ͥͬ̾a̴͕̲̒̒͌̋ͪl̨͎̰̘͉̟ͤ̀̈̚͜g͕͔̤͖̟̒͝ͅo̵̡̡̼͚̐ͯ̅ͪ̆ͣ̚', style: style.copyWith(letterSpacing: 1.0)),
            const SizedBox(height: 100.0),
            Text('Z͉̳̺ͥͬ̾a̴͕̲̒̒͌̋ͪl̨͎̰̘͉̟ͤ̀̈̚͜g͕͔̤͖̟̒͝ͅo̵̡̡̼͚̐ͯ̅ͪ̆ͣ̚', style: style.copyWith(letterSpacing: 2.0)),
            const SizedBox(height: 100.0),
          ],
        ),
      ),
    );
  }
}
```
</details>

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.3.10, on macOS 13.1 22C65 darwin-arm, locale en-GB)
    • Flutter version 3.3.10 on channel stable at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (4 weeks ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.74.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (4 available)
    • Pixel 7 (mobile)       • 28291FDH2001SA                       • android-arm64  • Android 13 (API 33)
    • iPhone 14 Pro (mobile) • 7256EA18-F2CB-4F96-9398-80DC7756A275 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 13.1 22C65 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 108.0.5359.124

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[!] Flutter (Channel master, 3.7.0-16.0.pre.12, on macOS 13.1 22C65 darwin-arm64, locale en-GB)
    • Flutter version 3.7.0-16.0.pre.12 on channel master at /Users/nexus/dev/sdks/flutters
    ! Warning: `flutter` on your path resolves to /Users/nexus/dev/sdks/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutters. Consider adding /Users/nexus/dev/sdks/flutters/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/nexus/dev/sdks/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutters. Consider adding /Users/nexus/dev/sdks/flutters/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 466cb54633 (47 minutes ago), 2023-01-10 00:47:24 -0500
    • Engine revision 716bb9172c
    • Dart version 3.0.0 (build 3.0.0-103.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.74.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (4 available)
    • Pixel 7 (mobile)       • 28291FDH2001SA                       • android-arm64  • Android 13 (API 33)
    • iPhone 14 Pro (mobile) • 7256EA18-F2CB-4F96-9398-80DC7756A275 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 13.1 22C65 darwin-arm64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 108.0.5359.124

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```
</details>
